src/unstable-temp/DownloadImageModal/OsConfiguration.tsx

```diff
@@ -259,7 +261,7 @@ export const OsConfiguration = ({
 					</Box>
 				)}
 				{(!isInitialDefault || !selectedOsType) && hasEsrVersions && (
-					<Box flex={2} ml={2}>
+					<Box flex={2} ml={hasMultipleCompatibleDeviceTypes ? 2 : undefined}>
```

Comment:
I would really prefer to reduce the conditionals. An extra pointer of how badly this scales is that when the OsTypeSelector is hidden, we end up with an orphan `mr={2}` in DeviceTypeSelector.
On top of that, this component will be converted to a purely data driven form, in which case we will not even be able to do this kind of conditionals for formatting.
Going with negative margins or even better using css grid sounds what we will need to do later anyway.

PS:We already have a DataGrid, which might fit here as well, but it needs to be able accept the margin as a parameter (and preserve the current value as a default).

Reply:
Ah I see, I wasn't aware that the conditionals had this much of an effect. Thank you for explaining, I'll reduce the conditionals 👍